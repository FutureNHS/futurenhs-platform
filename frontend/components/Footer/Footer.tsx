import React from "react";

import Link from "next/link";
import { Footer as NhsFooter } from "nhsuk-react-components";

interface Props {
  title: string;
  href: string;
}

export const FooterListItem: React.FC<Props> = ({ title, href }) => (
  <li className="nhsuk-footer__list-item">
    <Link href={href} passHref>
      <a className="nhsuk-footer__list-item-link">{title}</a>
    </Link>
  </li>
);

const Footer = () => {
  return (
    <NhsFooter>
      <NhsFooter.List>
        <FooterListItem title="About" href="#" />
        <FooterListItem title="Terms and Conditions" href="#" />
        <FooterListItem title="Accessibility Statement" href="#" />
        <FooterListItem title="Privacy Policy" href="#" />
      </NhsFooter.List>
      <NhsFooter.Copyright>
        &copy; All content is available under the Open Government Licence v3.0,
        except where otherwise stated.
      </NhsFooter.Copyright>
    </NhsFooter>
  );
};

export default Footer;
