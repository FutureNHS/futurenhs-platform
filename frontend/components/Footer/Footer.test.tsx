import React from "react";

import { render } from "@testing-library/react";

import Footer, { FooterListItem } from "./Footer";

describe(Footer, () => {
  it("renders correctly", () => {
    const container = render(<Footer />);

    expect(container.asFragment()).toMatchSnapshot();
  });
});

describe(FooterListItem, () => {
  it("renders correctly with props", () => {
    const container = render(
      <FooterListItem title="testTitle" href="testHref" />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
