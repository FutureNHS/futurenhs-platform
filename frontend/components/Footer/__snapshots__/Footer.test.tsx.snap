// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="nhsuk-footer"
    >
      <div
        class="nhsuk-width-container"
      >
        <h2
          class="nhsuk-u-visually-hidden"
        >
          Support links
        </h2>
        <ul
          class="nhsuk-footer__list"
        >
          <li
            class="nhsuk-footer__list-item"
          >
            <a
              class="nhsuk-footer__list-item-link"
              href="/#"
            >
              About
            </a>
          </li>
          <li
            class="nhsuk-footer__list-item"
          >
            <a
              class="nhsuk-footer__list-item-link"
              href="/#"
            >
              Terms and Conditions
            </a>
          </li>
          <li
            class="nhsuk-footer__list-item"
          >
            <a
              class="nhsuk-footer__list-item-link"
              href="/#"
            >
              Accessibility Statement
            </a>
          </li>
          <li
            class="nhsuk-footer__list-item"
          >
            <a
              class="nhsuk-footer__list-item-link"
              href="/#"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
        <p
          class="nhsuk-footer__copyright"
        >
          Â© All content is available under the Open Government Licence v3.0, except where otherwise stated.
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`FooterListItem renders correctly 1`] = `
<DocumentFragment>
  <li
    class="nhsuk-footer__list-item"
  >
    <a
      class="nhsuk-footer__list-item-link"
      href="/testHref"
    >
      test
    </a>
  </li>
</DocumentFragment>
`;

exports[`FooterListItem renders correctly with props 1`] = `
<DocumentFragment>
  <li
    class="nhsuk-footer__list-item"
  >
    <a
      class="nhsuk-footer__list-item-link"
      href="/testHref"
    >
      testTitle
    </a>
  </li>
</DocumentFragment>
`;
